otherlibs/stdune-unstable/user_error.mli

```diff
@@ -1,40 +1,55 @@
 (** Error meant for humans *)
 
-module Annot : sig
-  type t
+(** A data structure carrying extra information that can be attached to errors.
+
+    This is mostly used to provide further context, such as location information
+    for the RPC. *)
+module Annotations : sig
+  module Entry : sig
```

Comment:
`Annotations` and `Annotations.Entry` feels like too much vocabulary for this feature, what about:

```
module Annotation : sig
   ...
end

module Annotations : sig
  ...
end
```

?